hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -737,7 +737,10 @@ public boolean next(List<Cell> outResult, ScannerContext scannerContext) throws
 
           case SEEK_NEXT_USING_HINT:
             Cell nextKV = matcher.getNextKeyHint(cell);
-            if (nextKV != null && comparator.compare(nextKV, cell) > 0) {
+            if (nextKV != null &&
+              ((!scan.isReversed() && comparator.compare(nextKV, cell) > 0)
```

Comment:
Let's only do one compare and record the result and use it in the if condition? Here we are on the critial read path, so let's try to avoid extra compare as much as possible.

Reply:
Java's && short-circuits, so the comparison is only ever actually run once. If the `isReversed` check (either the normal or the negated) fails, we don't do the associated comparison.

However, I agree that it would be clearer that only one comparison is performed if we store the result. Updated.

I also removed the extraneous `@throws` doc tags; they were left over when copying from `TestFuzzyRowFilterEndToEnd`.

P. S. Do you know why GitHub says there were test failures, but the linked "details" page doesn't report any (besides a known failure)?

Reply:
OK, thank you for making it more clear.

On the broken PR state, it is because this issue

https://issues.apache.org/jira/browse/SUREFIRE-1821

We enable flaky rerun feature and once there are some flaky tests, the xml report will be broken and fail the report generation stage of the jenkins job, which will finally lead a broken state here.

Still trying to solve the problem but not a blocker for merging here.